torch/onnx/_internal/onnxruntime.py

```diff
@@ -828,56 +928,100 @@ def compile(self, graph_module: torch.fx.GraphModule, args) -> torch.fx.GraphMod
     def __call__(
         self, graph_module: torch.fx.GraphModule, args
     ) -> torch.fx.GraphModule:
-        """Interface for dynamo to compile a graph module.
+        """If ``OrtBackendOptions.use_aot_autograd`` is ``True``, the `auto_autograd` compiler
+        will be invoked, wrapping this ``OrtBackend`` instance's ``compile`` method. Otherwise,
+        the ``compile`` method is invoked directly."""
+        if self._options.use_aot_autograd:
+            from torch._dynamo.backends.common import aot_autograd
+
+            return aot_autograd(
+                fw_compiler=self.compile,
+                partition_fn=min_cut_rematerialization_partition,
+                decompositions=self._resolved_onnx_exporter_options.decomposition_table,
+            )(graph_module, args)
 
-        All backends/compilers are passed into dynamo as
-        callables following this signature (mapping from
-        (torch.fx.GraphModule, args) to torch.fx.GraphModule).
-        """
         return self.compile(graph_module, args)
 
+    __instance_cache: Final[List["OrtBackend"]] = []
```

Comment:
nit: should there be a cache limit? I'm not sure what's the use case scenario for backend caching. Just from a perspective that backends themselves cache compiled graphs and sessions which might be large.

Reply:
We don't realistically think there'd be more than a few in a given process, but I agree, I should assert that at least. I will update.

This stems from the original design always using a singleton instance of `OrtBackend`, which was fine, except you could not configure it 😄 . Allowing configurability however, means we need to cache based on that, or do not cache at all.

I have been working through this with @wschin this week, and he thought it better to keep some manner of caching for now.

A followup would be to improve this deeper in `OrtBackend` such that creating instances of them are not costly, and they can reuse cached `InferenceSession`s instead. This would eliminate the need to cache `OrtBackend` at all.

Reply:
I picked 8. The `assert` directs the user to this comment thread.